src/main/java/com/exactpro/epfast/template/xml/AbstractReferenceImpl.java

```diff
@@ -0,0 +1,62 @@
+/******************************************************************************
+ * Copyright 2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.exactpro.epfast.template.xml;
+
+import com.exactpro.epfast.template.Reference;
+
+public abstract class AbstractReferenceImpl implements Reference {
```

Comment:
It can be extended from `AbstractNameapaceProvider` (or from delegating in the worst case)